src/js/components/AppPageComponent.jsx

```diff
@@ -52,74 +56,196 @@ var AppPageComponent = React.createClass({
     return {
       activeViewIndex: 0,
       activeTabId: activeTabId,
-      tabs: tabs
+      activeTaskId: null,
+      app: AppsStore.getCurrentApp(appId),
+      tabs: tabs,
+      fetchState: States.STATE_LOADING
     };
   },
 
+  componentWillMount: function () {
+    AppsStore.on(AppsEvents.CHANGE, this.onAppChange);
+    AppsStore.on(AppsEvents.REQUEST_APP_ERROR, this.onAppRequestError);
+    AppsStore.on(AppsEvents.SCALE_APP_ERROR, this.onScaleAppError);
+    AppsStore.on(AppsEvents.RESTART_APP_ERROR, this.onRestartAppError);
+    AppsStore.on(AppsEvents.DELETE_APP_ERROR, this.onDeleteAppError);
+    AppsStore.on(AppsEvents.DELETE_APP, this.onDeleteAppSuccess);
+  },
+
+  componentWillUnmount: function () {
+    AppsStore.removeListener(AppsEvents.CHANGE,
+      this.onAppChange);
+    AppsStore.removeListener(AppsEvents.REQUEST_APP_ERROR,
+      this.onAppRequestError);
+    AppsStore.removeListener(AppsEvents.SCALE_APP_ERROR,
+      this.onScaleAppError);
+    AppsStore.removeListener(AppsEvents.RESTART_APP_ERROR,
+      this.onRestartAppError);
+    AppsStore.removeListener(AppsEvents.DELETE_APP_ERROR,
+      this.onDeleteAppError);
+    AppsStore.removeListener(AppsEvents.DELETE_APP,
+      this.onDeleteAppSuccess);
+  },
+
```

Comment:
@rcorral  we could even extract this pattern into a `EventBinder` [mixin](https://facebook.github.io/react/docs/reusable-components.html#mixins). But I'd suggest to make this into its own refactoring task later on.